plot.py

```diff
@@ -68,12 +69,104 @@ class Plantation(ModelSQL, ModelView):
     parcels = fields.One2Many('agronomics.parcel', 'plantation', "Parcel")
     plantation_year = fields.Integer("Plantation Year")
     plantation_owner = fields.Many2One('party.party', "Plantation Owner")
+    varieties = fields.Function(fields.Char('Varieties'), 'get_varieties',
+        searcher='search_varieties')
+    do = fields.Function(fields.Char('DO'), 'get_all_do', searcher='search_do')
+    remaining_quantity = fields.Function(
+        fields.Float("Remainig Quantity", digits=(16, 2)),
+        'get_remaining_quantity', searcher='search_remaining_quantity')
 
     def get_rec_name(self, name):
         if self.code:
             return self.code
         return self.name
 
+    def get_all_do(self, name):
+        do = []
+        for y in self.parcels:
+            do += [x.name for x in y.denomination_origin]
+        return ",".join(list(set(do)))
+
+    def get_varieties(self, name):
+        varieties = []
+        for y in self.parcels:
+            if y.variety:
+                varieties += [y.variety and y.variety.name ]
```

Comment:
ho simplificaria:

```
        varieties = []
        for y in self.parcels:
            if y.variety:
                varieties += [y.variety and y.variety.name ]
```

per:

```
varieties = [y.variety.name for for y in self.parcels if y.variety]
```

Tampoc caldria si fas un "if variety" a dins del condicional tonar a fer y.variety and y.variety.name